core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -1301,7 +1307,18 @@ class TimeSeriesShard(val ref: DatasetRef,
     }
 
     val currentTime = System.currentTimeMillis()
-    shardStats.ingestionPipelineLatency.record(currentTime - container.timestamp)
+
+    // Account for clock skew, otherwise exceptions are thrown when negative latencies are recorded.
```

Comment:
Interesting problem (even more difficult to solve). I am guessing the currentTimeMillis is epoch time ?
Good idea to track it with metric. This is probably a task for kube team to arrest the drift, but we might need to provide evidence for the same. Should we log some instances of currentTime as well as container.timestamp to have concrete evidence for them ?

Reply:
It might be an expected behavior with the clocks getting synced periodically. I am assuming this dpwsry happen very frequently.